external-crates/move/crates/move-compiler/src/linters/mod.rs

```diff
@@ -17,16 +20,31 @@ pub enum LintLevel {
 
 pub const ALLOW_ATTR_CATEGORY: &str = "lint";
 pub const LINT_WARNING_PREFIX: &str = "Lint ";
+pub const BOOL_COMPARISON_FILTER_NAME: &str = "bool_comparison";
+
+pub const LINTER_DEFAULT_DIAG_CODE: u8 = 1;
+
+pub enum LinterDiagCategory {
+    UnnecessaryBoolComparison,
```

Comment:
I know you were likely copying the sui linter, but we didn't really do a great job splitting things correctly between category and code. 
I would imagine the category to be something like Unnecessary Operation? Then the code would be Bool Comparison.